src/cryptonote_core/miner.cpp

```diff
@@ -365,6 +398,11 @@ namespace cryptonote
         misc_utils::sleep_no_w(100);
         continue;
       }
+      else if( m_is_background_mining_enabled )
+      {
+        misc_utils::sleep_no_w(m_miner_extra_sleep);
```

Comment:
Hmm, that sleep is there is how I throttle the miner threads. 
The BACKGROUND_MINING_MINER_MONITOR_INVERVAL constant is the time between when the monitoring thread re-checks miner cpu usage. Right now I think I made them the same, but I think in the future the amount of time between checking whether to start will be greater than checking whether to stop. 

The mutex version is definitely an interesting idea, and I have no problem refactoring to whatever you guys agree is better.